modules/wallet/database.go

```diff
@@ -38,91 +68,115 @@ func dbDelete(b *bolt.Bucket, key interface{}) error {
 	return b.Delete(encoding.Marshal(key))
 }
 
-// dbPutHistoricClaimStart stores the number of siafunds corresponding to the
-// specified SiafundOutputID. The wallet builds a mapping of output -> value
-// so that it can determine the value of siafund outputs that were not created
-// by the wallet.
+// dbForEach is a helper function for iterating over a bucket and calling fn
+// on each entry. fn must be a function with two parameters. The key/value
+// bytes of each bucket entry will be unmarshalled into the types of fn's
+// parameters.
+func dbForEach(b *bolt.Bucket, fn interface{}) error {
+	// check function type
+	fnVal, fnTyp := reflect.ValueOf(fn), reflect.TypeOf(fn)
+	if fnTyp.Kind() != reflect.Func || fnTyp.NumIn() != 2 {
+		panic("bad fn type: needed func(key, val), got " + fnTyp.String())
+	}
+
+	return b.ForEach(func(keyBytes, valBytes []byte) error {
+		key, val := reflect.New(fnTyp.In(0)), reflect.New(fnTyp.In(1))
+		if err := encoding.Unmarshal(keyBytes, key.Interface()); err != nil {
+			return err
+		} else if err := encoding.Unmarshal(valBytes, val.Interface()); err != nil {
+			return err
+		}
+		fnVal.Call([]reflect.Value{key.Elem(), val.Elem()})
+		return nil
+	})
+}
+
+// Type-safe wrappers around the db helpers
+
 func dbPutHistoricClaimStart(tx *bolt.Tx, id types.SiafundOutputID, c types.Currency) error {
 	return dbPut(tx.Bucket(bucketHistoricClaimStarts), id, c)
 }
-
-// dbGetHistoricClaimStart retrieves the number of siafunds corresponding to
-// the specified SiafundOutputID. The wallet uses this mapping to determine
-// the value of siafund outputs that were not created by the wallet.
 func dbGetHistoricClaimStart(tx *bolt.Tx, id types.SiafundOutputID) (c types.Currency, err error) {
 	err = dbGet(tx.Bucket(bucketHistoricClaimStarts), id, &c)
 	return
 }
 
-// dbPutHistoricOutput stores the number of coins corresponding to the
-// specified OutputID. The wallet builds a mapping of output -> value so that
-// it can determine the value of outputs that were not created
-// by the wallet.
 func dbPutHistoricOutput(tx *bolt.Tx, id types.OutputID, c types.Currency) error {
 	return dbPut(tx.Bucket(bucketHistoricOutputs), id, c)
 }
-
-// dbGetHistoricOutput retrieves the number of coins corresponding to the
-// specified OutputID. The wallet uses this mapping to determine the value of
-// outputs that were not created by the wallet.
 func dbGetHistoricOutput(tx *bolt.Tx, id types.OutputID) (c types.Currency, err error) {
```

Comment:
The docstrings would have no value. It's plainly obvious that they are type-safe wrappers around `dbPut`, `dbGet`, etc. Those functions all have good docstrings. Furthermore, the purpose of each bucket is now explained at the top of the file. There's nothing to say in the docstring other than to describe what bucket and types it operates on -- which is given in the function definition.
Docstrings are always required on exported functions, but in rare cases, they can be omitted on unexported functions. The [stubs in the renter tests](https://github.com/NebulousLabs/Sia/blob/master/modules/renter/renter_test.go#L167) are a good example of where a single docstring makes more sense than docstrings on each individual function.